tests/phpunit/tests/option/option.php

```diff
@@ -118,6 +118,42 @@ public function test_get_option_notoptions_cache() {
 		$this->assertSame( 0, $after - $before );
 	}
 
+	/**
+	 * @ticket 58277
+	 *
+	 * @covers ::get_option
+	 */
+	public function test_get_option_notoptions_set_cache() {
+		get_option( 'invalid' );
+
+		$before = get_num_queries();
+		$value  = get_option( 'invalid' );
+		$after  = get_num_queries();
+
+		$notoptions = wp_cache_get( 'notoptions', 'options' );
+
+		$this->assertSame( 0, $after - $before, 'No more queries should be run on second run on function' );
+		$this->assertArrayHasKey( 'invalid', $notoptions, 'Check that option name is in notoptions cache' );
```

Comment:
```suggestion
		$this->assertSame( 0, $after - $before, 'The notoptions cache was not hit on the second call to `get_option()`.' );
		$this->assertArrayHasKey( 'invalid', $notoptions, 'The "invalid" option should be in the notoptions cache.' );
```
Some improvements to the assertion failure messages.

Reply:
Additionally, we should add this before the `$this->assertArrayHasKey()` line:

```php
$this->assertIsArray( $notoptions, 'The notoptions cache should be set.' );
```
If source changes ever caused the notoptions cache to not be set, `$notoptions` would be `false`, and `$this->ArrayHasKey()` would error out.